### PR TITLE
Feat: List changed files by git repo manager

### PR DESCRIPTION
**Pull request by emranbm**

### CODE REVIEW

File: src/gitrepo/manager.go [hunk 1]
```diff
@@ -9,18 +9,44 @@ import (
...
 func (m *gitRepoManager) Checkout(refName string) {
-	utils.PanicError(m.worktree.Checkout(&git.CheckoutOptions{
+	worktree := utils.GetResultOrPanicError(m.repo.Worktree())
+	utils.PanicError(worktree.Checkout(&git.CheckoutOptions{
 		Branch: plumbing.ReferenceName("refs/heads/" + refName),
 		Force:  true,
 	}))
 }
+
+func (m *gitRepoManager) ListChangedFiles(fromRevision string, toRevision string) []string {
```
Comment (Adibov) [inline src/gitrepo/manager.go:29]: How about `srcRevision` and `destRevision` for this method's arguments?

File: src/gitrepo/manager.go [hunk 2]
```diff
@@ -9,18 +9,44 @@ import (
...
+	fromHash := utils.GetResultOrPanicError(m.repo.ResolveRevision(plumbing.Revision(fromRevision)))
```
Comment (Adibov) [inline src/gitrepo/manager.go:30]: Same as `srcHash` and etc.

File: src/gitrepo/manager.go (hunk 1 above)
Comment (Adibov) [inline src/gitrepo/manager.go:29]: Doesn't the _git_ package have some entity for revisions to use instead?

File: src/gitrepo/manager.go [hunk 3]
```diff
@@ -9,18 +9,44 @@ import (
...
+	toHash := utils.GetResultOrPanicError(m.repo.ResolveRevision(plumbing.Revision(toRevision)))
+	toCommit := utils.GetResultOrPanicError(m.repo.CommitObject(*toHash))
+	patch := utils.GetResultOrPanicError(fromCommit.Patch(toCommit))
+	filePathsSet := make(map[string]bool)
+	for _, filePatch := range patch.FilePatches() {
+		fromFile, toFile := filePatch.Files()
+		if fromFile != nil {
+			filePathsSet[fromFile.Path()] = true
+		}
+		filePathsSet[toFile.Path()] = true
```
Comment (Adibov) [inline src/gitrepo/manager.go]: From documentation:
```
	// Files returns the from and to Files, with all the necessary metadata to
	// about them. If the patch creates a new file, "from" will be nil.
	// If the patch deletes a file, "to" will be nil.
```

So, can't toFile also be nil?

File: src/gitrepo/manager.go [hunk 4]
```diff
@@ -9,18 +9,44 @@ import (
...
+	}
+
+	changedFilePaths := make([]string, len(filePathsSet))
+
+	i := 0
+	for path := range filePathsSet {
```
Comment (Adibov) [inline src/gitrepo/manager.go:49]: Better to use this format instead:
```golang
for index, element := range array {
    ...
}
```

**Review by Adibov — COMMENTED** [review]
Welldone!

File: src/gitrepo/manager.go (hunk 1 above)
Comment (emranbm) [reply src/gitrepo/manager.go:29]: Great suggestion! Thanks. I'll open a separate PR for the current names as well.

File: src/gitrepo/manager.go (hunk 3 above)
Comment (emranbm) [reply src/gitrepo/manager.go]: You're absolutely right. Thanks.

File: src/gitrepo/manager.go (hunk 1 above)
Comment (emranbm) [reply src/gitrepo/manager.go:29]: Seems the rename is done in another [PR](https://github.com/emranbm/semangit/pull/3)

File: src/gitrepo/manager.go (hunk 4 above)
Comment (emranbm) [reply src/gitrepo/manager.go:49]: Why? I just need the keys.

File: src/gitrepo/manager.go (hunk 1 above)
Comment (emranbm) [reply src/gitrepo/manager.go:29]: Based on our discussion, I added a doc to emphasize revision meaning. Please review it again and resolve the thread if that's ok.

**Review by Adibov — CHANGES_REQUESTED** [review]
Approving review!

**Review by Adibov — APPROVED** [review]

**Pull request merged**